src/language-handlebars/printer-glimmer.js

```diff
@@ -489,29 +579,37 @@ function blockStatementHasElse(node) {
   return isNodeOfSomeType(node, ["BlockStatement"]) && node.inverse;
 }
 
-function printProgram(path, print) {
+function printProgram(path, print, options) {
   const node = path.getValue();
 
   if (blockStatementHasOnlyWhitespaceInProgram(node)) {
     return "";
   }
 
   const program = path.call(print, "program");
-  return indent(concat([hardline, program]));
+
+  if (options.htmlWhitespaceSensitivity !== "strict") {
+    return indent(concat([hardline, program]));
+  }
+
+  return indent(program);
 }
 
-function printInverse(path, print) {
+function printInverse(path, print, options) {
   const node = path.getValue();
 
   const inverse = path.call(print, "inverse");
-  const parts = concat([hardline, inverse]);
+  const parts =
+    options.htmlWhitespaceSensitivity !== "strict"
+      ? [hardline, inverse]
+      : [inverse];
 
   if (blockStatementHasElseIf(node)) {
-    return parts;
+    return concat(parts);
   }
 
   if (blockStatementHasElse(node)) {
-    return concat([printElseBlock(node), indent(parts)]);
+    return concat([printElseBlock(node, options), indent(concat(parts))]);
   }
```

Comment:
```suggestion
  const printed =
    options.htmlWhitespaceSensitivity !== "strict"
      ? concat([hardline, inverse])
      : inverse;

  if (blockStatementHasElseIf(node)) {
    return printed;
  }

  if (blockStatementHasElse(node)) {
    return concat([printElseBlock(node, options), indent(printed)]);
  }
```

Reply:
I manually applied your suggestion (it was not possible anymore after the rebase)